src/main/java/seedu/address/logic/commands/ImportCommand.java

```diff
@@ -0,0 +1,126 @@
+package seedu.address.logic.commands;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.assignment.Assignment;
+import seedu.address.model.person.Address;
+import seedu.address.model.person.Email;
+import seedu.address.model.person.Github;
+import seedu.address.model.person.Name;
+import seedu.address.model.person.Person;
+import seedu.address.model.person.Phone;
+import seedu.address.model.person.Telegram;
+import seedu.address.model.tag.Tag;
+
+/**
+ * Import command to load data from specified CSV file location.
+ */
+public class ImportCommand extends Command {
+
+    public static final String COMMAND_WORD = "import";
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Imports person data from a CSV file.\n"
+        + "Parameters: FILE_PATH\n"
+        + "Example: " + COMMAND_WORD + " data/persons.csv";
+
+    private final String csvFilePath;
+
+    public ImportCommand(String csvFilePath) {
+        this.csvFilePath = csvFilePath;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        try (BufferedReader br = new BufferedReader(new FileReader(csvFilePath))) {
+            String line;
+            List<Person> newPersons = new ArrayList<>();
+
+            // Skip header line
+            br.readLine();
+
+            while ((line = br.readLine()) != null) {
+                String[] fields = line.split(",");
+                if (fields.length < 9) {
+                    throw new CommandException("Invalid CSV format.");
+                }
+
+                try {
```

Comment:
If possible, it would be better if the try catch block could be split to adhere to SLAP principles